product_docs/docs/pgd/5/admin-tpa/upgrading_major_rolling.mdx

```diff
@@ -0,0 +1,537 @@
+---
+title: Performing a Postgres major version rolling upgrade on a PGD cluster built with TPA
+navTitle: Upgrade Postgres major version
+deepToC: true
+---
+
+## Overview
+
+You can perform a major version upgrade of Postgres on a Postgres Distributed (PGD) cluster deployed by Trusted Postgres Architect (TPA).
+
+The rolling upgrade process allows updating individual cluster nodes to a new major Postgres version while maintaining cluster availability and operational continuity. This approach minimizes downtime and ensures data integrity by allowing the rest of the cluster to remain operational as each node is upgraded sequentially. 
+
+We recommend upgrading your Postgres cluster to a newer major version to access improved features, performance enhancements, and security updates. These step-by-step instructions and a worked example help to provide a smooth and controlled upgrade process.
+
+### Plan the upgrade
+
+The cluster must be available throughout the process (that is, a *rolling* upgrade). To achieve continuous cluster availability, you must [perform a planned switchover](#performing-a-planned-switchover) of the [write leader](https://www.enterprisedb.com/docs/pgd/latest/terminology/#write-leader) node before upgrading it so that a write leader is always available. 
+
+You can see which node is the write leader by connecting to the PGD cluster (for example, using SSH) and running the `pgd show-groups` command:
+
+```bash
+sudo -u postgres pgd show-groups
+```
+
+To keep the number of planned [switchovers](https://www.enterprisedb.com/docs/pgd/latest/cli/command_ref/pgd_switchover/) to a minimum, upgrade the current write leader last in your upgrade order, so you have to perform only [one planned switchover](#performing-a-planned-switchover) during the upgrade process.
```

Comment:
```suggestion
To keep the number of planned [switchovers](https://www.enterprisedb.com/docs/pgd/latest/cli/command_ref/pgd_switchover/) to a minimum, put the current write leader last in your upgrade order. This will allow you to only perform [one planned switchover](#performing-a-planned-switchover) during the upgrade process.
```